champ/node/validation/block.rs

```diff
@@ -27,6 +30,23 @@ fn verify_transactions() -> Result<()> {
 }
 
 // Verifies the block height and previous block
-fn verify_previous_block() -> Result<()> {
-    unimplemented!()
+fn verify_previous_block(new_block: &Block, prev_block: &Block) -> Result<()> {
+    let new_data = new_block.data.as_ref().ok_or_else(|| anyhow!("block data not found"))?;
+    let prev_data = prev_block
+        .data
+        .as_ref()
+        .ok_or_else(|| anyhow!("block data not found"))?;
+
+    if new_data.height == 0 {
+        // if new_block is the first block in the chain
+        // TODO: add some magic I guess
```

Comment:
maybe `get_block_by_height` could return an option if there is no block with a given height and if that option is none we could skip verifying the previous block entirely. Currently this code wouldn't get called anyway since previous block has to exists, so this should return an error instead of Ok